src/drivers/sqs/Queue.php

```diff
@@ -0,0 +1,206 @@
+<?php
+
+namespace yii\queue\sqs;
+
+use yii\base\NotSupportedException;
+use yii\queue\cli\Queue as CliQueue;
+use yii\queue\cli\Signal;
+use \Aws\Sqs\SqsClient;
+use Aws\Credentials\CredentialProvider;
+
+/**
+ * SQS Queue
+ *
+ * @author Manoj Malviya <manojm@girnarsoft.com>
+ */
+class Queue extends CliQueue
+{
+    /**
+    * @var SqsClient
+    */
+    private $_client;
+
+    /**
+     * The SQS url.
+     * @var string
+     */
+    public $url;
+
+    /**
+     * aws access key
+     * @var string
+     */
+    public $key = '';
+
+    /**
+     * aws secret
+     * @var string
+     */
+    public $secret = '';
+
+    /**
+     * region where queue is hosted.
+     * @var string
+     */
+    public $region = '';
+
+    /**
+     * API version
+     * @var string
+     */
+    public $version = 'latest';
+
+    /**
+     * @var string command class name
+     */
+    public $commandClass = Command::class;
+
+    /**
+     * @inheritdoc
+     */
+     public function init()
+     {
+         parent::init();
+     }
+
+    /**
+     * Runs all jobs from queue.
+     */
+    public function run()
+    {
+        while (!Signal::isExit() && ($payload = $this->getPayload())) {
+            list($ttr, $message) = explode(';', $payload['Body'], 2);
+
+            $this->reserve($payload, $ttr); //reserve it so it is not visible to another worker till ttr
+
+            if ($this->handleMessage(null, $message, $ttr, 1)) {
+                //if handled then remove from queue
+                $this->release($payload);
+            }
+        }
+    }
+
+    /**
+     * Listens to get new jobs.
+     */
+    public function listen()
+    {
+        $this->run();        
```

Comment:
I think this method should do the same as in other queue drivers. $queue->listen() should run an infinite loop with timeout (sleep) to use as a daemon, regardless of payload existence.